app/controllers/api/providers_controller.rb

```diff
@@ -85,6 +85,15 @@ def import_vm_resource(type, id = nil, data = {})
       end
     end
 
+    def options
+      req_class_name = @req.to_hash[:fullpath]
+                           .split('?').last.split('&')                        # '?' for query, '&' connects parameters
+                           .select { |param| param.starts_with?("ems_type") } # 'parameters with "ems_type" name
+                           .first.split('=').last                             # choosing only the first occuracne.
+      ems_class = ExtManagementSystem.descendants.select { |ems| ems.ems_type == req_class_name }.first
```

Comment:
@enoodle ah, gotcha! yeah, then `descendant_get` won't work.

this is just nit but you could do `ExtManagementSystem.descendants.find { |ems| ems.ems_type == params[:ems_type] }`